force-app/main/default/classes/DailyIterator_TEST.cls

```diff
@@ -4,19 +4,111 @@ public with sharing class DailyIterator_TEST {
     private static void shouldAssignRule() {
         RecurrenceRule rRule = new RecurrenceRule();
         DailyIterator iterator = new DailyIterator(rRule);
-        System.assertEquals(
-            rRule,
-            iterator.recurrenceRule,
-            'Should assign rule on construction.'
-        );
+        System.assertEquals(rRule, iterator.rRule, 'Should assign rule on construction.');
+    }
+
+    /* FUNCTIONAL TESTS */
+
+    @IsTest
+    private static void shouldIterateEveryDay5Times() {
+        Datetime datetimeStart = Datetime.newInstance(2020, 9, 1, 14, 0, 0); // Sept 1 2020, 2PM
+        RecurrenceRule rRule = new RecurrenceRule().withRuleString('FREQ=DAILY;COUNT=5'); // every day, 5 times
+        List<Datetime> expected = new List<Datetime>{
+            Datetime.newInstance(2020, 9, 1, 14, 0, 0),
+            Datetime.newInstance(2020, 9, 2, 14, 0, 0),
+            Datetime.newInstance(2020, 9, 3, 14, 0, 0),
+            Datetime.newInstance(2020, 9, 4, 14, 0, 0),
+            Datetime.newInstance(2020, 9, 5, 14, 0, 0)
+        };
+        DailyIterator iterator = new DailyIterator(rRule);
+
+        Test.startTest();
+        List<Datetime> actual = iterator.iterate(datetimeStart);
+        Test.stopTest();
+
+        System.assertEquals(expected, actual);
```

Comment:
Never hurts to have them, that way if the assert ever fails we can tell our future selves what we were expecting. Sometimes it is pretty clear with the method name and the expected results but in case where we are just asserting true or false it can be especially useful.

Reply:
`List of Dates returned from the iterator was not as expected.`